src/main/scala/org/broadinstitute/dsde/firecloud/HttpClient.scala

```diff
@@ -41,17 +43,24 @@ class HttpClient (requestContext: RequestContext) extends Actor
 
   override def receive: Receive = {
 
-    case PerformExternalRequest(externalRequest: HttpRequest) =>
-      createResponseFutureFromExternalRequest(requestContext, externalRequest)
+    case PerformExternalRequest(requestCompression: Boolean, externalRequest: HttpRequest) =>
+      createResponseFutureFromExternalRequest(requestCompression, requestContext, externalRequest)
 
   }
 
   def createResponseFutureFromExternalRequest(
+      requestCompression: Boolean,
       requestContext: RequestContext,
       externalRequest: HttpRequest): Unit = {
     val pipeline: HttpRequest => Future[HttpResponse] =
-      authHeaders(requestContext) ~> addHeader(Cookie(requestContext.request.cookies)) ~>
-        logRequest(log) ~> sendReceive
+      requestCompression match {
+        case true =>
+          authHeaders(requestContext) ~> addHeaders(Cookie(requestContext.request.cookies), `Accept-Encoding`(gzip)) ~>
+            logRequest(log) ~> sendReceive ~> decode(Deflate)
```

Comment:
does `decode(Deflate)` work? Should this be `decode(Gzip)` instead? (here and elsewhere)